app/views/question_sets/_form.html.erb

```diff
@@ -26,7 +26,62 @@
     <% end %>
   </div>
 
+  <!-- check if current_user's courses have any questions
+        and allow user to filter those questions by searching tags.-->
+  <% if @questions.present? %>
+    <!-- add "search bar" for tags and filter questions shown based on
+          tag entered -->
+
+    <input type="text" id="tagSearchInput" onkeyup="qSetQTagSearch()" placeholder="Search for question tags...">
+
+    <table id="questionTagTable" border="1">
+      <tr>
+        <th></th>
+        <th style="min-width:250px">Body</th>
+        <th style="min-width:100px">Tags</th>
+      </tr>
+      <!-- loop thru questions from controller
+            (questions associated with courses the current_user is
+            a professor or assistant for) -->
+      <% @questions.each do |q| %>
+        <tr>
+          <td><%= check_box_tag 'question_ids[]', q.id, false, :id => 'question_ids_#{q.id}' %>
+          <td><%= q.body %></td>
+          <td><%= q.tags.join(", ") %></td>
+        </tr>
+      <% end %>
+    </table>
+  <% else %>
+    <!-- current_user's courses do not have any questions -->
+    <p>There are no questions associated with the courses you teach. Please create a question to add to the question set.</p>
+  <% end %>
+
   <div class="actions">
     <%= form.submit %>
   </div>
 <% end %>
+
+<script>
```

Comment:
The methods are a bit different, though, based on which column in the table the tags are in (1 or 2).